code/modules/power/hydroelectric.dm

```diff
@@ -0,0 +1,240 @@
+//Hydroelectric power by Sieghardt Meldurson. Please message me in case of issues.
+//The Hydro Turbine itself
+#define HYDRO_MAX_DIST 100
+/obj/machinery/power/hydroelectric
+	name = "hydroelectric turbine"
+	desc = "A hydroelectric generator able to store water and then release it to generate electricity. This one seems to be important to the structural integrity around and is attached directly to the dam's frame."
+	icon = 'icons/obj/machines/thermoelectric.dmi'
+	icon_state = "circ-unassembled"
+	density = 1
+	anchored = 1
+	use_power = NO_POWER_USE
+	idle_power_usage = 0
+	active_power_usage = 0
+	health = 10
+	var/malfstate = 0
+	var/id = 0
+	var/powrate = 10
+	var/trickleout = 2
+	var/obj/machinery/power/hydroelectric_control/control = null
+
+/obj/machinery/power/hydroelectric/drain_power()
+	return -1
+
+/obj/machinery/power/hydroelectric/update_icon()
+	cut_overlays()
+	if(!control)
+		return
+	if(malfstate != 0)
+		add_overlay(image('icons/obj/machines/thermoelectric.dmi', icon_state = "circ-hot", layer = FLY_LAYER))
+	else if (control.working == 1)
+		add_overlay(image('icons/obj/machines/thermoelectric.dmi', icon_state = "circ-run", layer = FLY_LAYER))
+	return
+
+/obj/machinery/power/hydroelectric/proc/Malfunction()
+	malfstate = rand(1,2)
+	if(malfstate == 1)
```

Comment:
Need to use bitflags instead of random values.

Reply:
The idea is that we'd have a random broken state. 1 and 2 are different broken states, with each requiring its own method of fixing.

Reply:
Yeah, I know. Use bitflags instead of a cryptic 1 and 2.

Reply:
Or, if you really can't do bitflags, use constants.